### PR TITLE
Add TradingView GAP 

### PR DESCRIPTION
**Pull request by ingf**
## Proposed changes

Add TradingView GAP

## Types of changes

What types of changes does your code introduce to BankrollHunter?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New script (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have tested the scripts with TradingView
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by lisangzhou — APPROVED** [review]
LGTM
Screen Shot 2020-10-06 at 11 22 27 AM

**Comment by donlingliang** [comment]
i am wondering how many gaps we could display though?

**Comment by ingf** [comment]
`length = input(300, minval=1, maxval=2000)`
默认显示 300 条 k 线中的缺口，最大可以设置到 2000 条 k 线

**Review by jchang274 — APPROVED** [review]

**Review by donlingliang — APPROVED** [review]

**Review by jchang274 — APPROVED** [review]

File: GAP/gap.pine [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+gap_start = high[length]
+gap_end = low[length - 1]
+gap_bull = false
+gap_bear = false
+inf_gap = 0.0
+sup_gap = 0.0
+sup_index = 0
+
+if barstate.islast
```
Comment (Coding-Fun) [inline GAP/gap.pine:20]: nit: pine script可以reverse logic然后return么 这样代码整体indentation会少些

File: GAP/gap.pine [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    for j = 1 to length
+        gap_bull := false
+        gap_bear := false
+        inf_gap := 0.0
+        sup_gap := 0.0
```
Comment (Coding-Fun) [inline GAP/gap.pine:25]: nit: 可以加个简单的注释 解释下sup和inf么

File: GAP/gap.pine [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        sup_index := j
```
Comment (Coding-Fun) [inline GAP/gap.pine:26]: sup_index为什么不从j-1开始取值？

**Review by kentio — APPROVED** [review]

**Pull request merged**